english/tests/tests.py

```diff
@@ -5,121 +5,292 @@
+
+
+class LessonsForTeacher(APITestCase):
+    fixtures = [
+        "english/tests/fixtures.json",
+    ]
+
+    def setUp(self):
+        self.student = User.objects.get(id=1)
+        self.teacher = User.objects.get(id=2)
+
+    def test_get_lessons_success(self):
+        self.client.force_login(user=self.teacher)
         expected_data = {
-            "pk": 2,
-            "class_name": 2,
-            "_status": "PROGRESS",
-            "time_start": "2022-07-04T01:14:43",
-            "time_end": "2022-07-30T01:14:46"
+            "pk": 1,
+            "class_name": "FirstClass",
+            "_status": "COMING",
+            "time_start": "2030-07-04T01:14:43",
+            "time_end": "2030-10-30T01:14:46"
         }
-        url = reverse("teacher-lessons")
+        url = reverse("teacher-lessons-list-create")
         response = self.client.get(url)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(len(response.data), 1)
         self.assertDictEqual(dict(response.data[0]), expected_data)
+
+    def test_get_lessons_failed(self):
+        self.client.force_login(user=self.student)
+        url = reverse("teacher-lessons-list-create")
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_get_one_lesson_success(self):
+        self.client.force_login(user=self.teacher)
+        expected_data = {
+            "pk": 1,
+            "class_name": "FirstClass",
+            "_status": "COMING",
+            "time_start": "2030-07-04T01:14:43",
+            "time_end": "2030-10-30T01:14:46"
+        }
+        url = "http://0.0.0.0:8000/english/teacher/lessons/1/"
+        response = self.client.get(url)
+        self.assertDictEqual(dict(response.data), expected_data)
+
+    def test_update_lesson_success(self):
+        self.client.force_login(user=self.teacher)
+        data = {
+            "time_start": "2030-07-04T01:14:43",
+            "time_end": "2032-10-30T01:14:46"
+        }
+        expected_data = {
+            "pk": 1,
+            "class_name": "FirstClass",
+            "_status": "COMING",
+            "time_start": "2030-07-04T01:14:43",
+            "time_end": "2032-10-30T01:14:46"
+        }
+        url = "http://0.0.0.0:8000/english/teacher/lessons/1/"
+        self.client.patch(url, data=data)
+        url = reverse("teacher-lessons-list-create")
+        response = self.client.get(url)
+        self.assertDictEqual(dict(response.data[0]), expected_data)
+
+    def test_update_lesson_failed(self):
+        self.client.force_login(user=self.teacher)
+        data = {
+            "time_start": "2050-07-04T01:14:43",
+            "time_end": "2030-10-30T01:14:46"
+        }
+        url = "http://0.0.0.0:8000/english/teacher/lessons/1/"
+        response = self.client.patch(url, data=data)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```

Comment:
Need to check the body. In the body, you will get the error message. Need to check this message